### PR TITLE
Initial Page

### PR DESCRIPTION
**Pull request by Roald-Schaum**
Page is Statically hosted

### CODE REVIEW

**Comment by ryleu** [comment]
SHC-ASTRA/rover-Basestation-Release#37

File: .gitignore
Comment (ryleu) [inline .gitignore:1]: we should add our output directory (currently `static/`, but please see my overall comment) to the gitignore so we don't push built files

File: README.md [hunk 1]
```diff
@@ -1 +1,14 @@
...
+To use the launch script,run the following command in the root folder:
+
+```bash
+. scripts/launch_frontend.sh
+```
+
+To launch the frontend manually, run the following command in rover-Basestation-Frontend:
+
+```bash
+npm run dev -- --open
```
Comment (ryleu) [inline README.md]: we want to build the files and ONLY build them. the built files should be placed in our output directory (again, see my main comment for this review)

File: README.md [hunk 2]
```diff
@@ -1 +1,14 @@
+# Testing
 
+Testing can be run either through the launch script, or manually.
+To use the launch script,run the following command in the root folder:
+
+```bash
+. scripts/launch_frontend.sh
```
Comment (ryleu) [inline README.md]: there shouldn't really be a frontend to launch, as the goal is to have it be static. we don't need any other scripts than build and maybe test and those can both be defined in `package.json`

File: colors.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+#0f1078
```
Comment (ryleu) [inline colors.txt]: whar

if this is
1. a color pallete: we should have those defined as variables in a `:root` element in some css file
2. some weird temporary file: get rid of it
3. something else: explain what the hell it is cuz i can't figure it out

File: eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    files: ["**/*.svelte"],
+
+    languageOptions: {
+	  parserOptions: {
+	    parser: ts.parser
+	  }
+	}
+  },
+  {
+    ignores: ["build/", ".svelte-kit/", "dist/"]
```
Comment (ryleu) [inline eslint.config.js]: `build/` in the ignore list should be whatever our output directory is (see main comment)

File: package.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
+	"name": "rover-basestation-frontend",
+	"version": "0.0.1",
+	"type": "module",
+	"scripts": {
+		"dev": "vite dev",
+		"build": "vite build",
```
Comment (ryleu) [inline package.json:7]: make sure the `build` command works properly and places the built files in the output directory

File: src/routes/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<style>
```
Comment (ryleu) [inline src/routes/+page.svelte]: looks good, but let's put our stylesheet in a different file (perhaps `style.css`?) and please test that we can get typescript working

File: static/.nojekyll
Comment (ryleu) [inline static/.nojekyll:1]: should not be committed

File: static/favicon.png
Comment (ryleu) [inline static/favicon.png:1]: should not be committed

**Review by ryleu — CHANGES_REQUESTED** [review]
we should consider changing the output directory to `build/`. the only directory i wanted to be named `static` was the one in -Release. i don't care what it's called in -Frontend, so sticking with the standard `build/` makes sense to me

when i make a commit to switch -Release to point to this version of -Frontend, i will also create a symlink from -Release/static to whatever you choose as the output directory

File: package.json (hunk 5 above)
Comment (Roald-Schaum) [reply package.json:7]: Configuration performed in svelte.config.js.

File: src/routes/+page.svelte [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<div>
+        <h1>Test</h1>
+        <p>This is pink</p>
+        <h2>I hope this works properly</h2>
+</div>
+
+<script>
+    import '../styles/global.css';
```
Comment (ryleu) [inline src/routes/+page.svelte]: use [`<link>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link) to include stylesheets. they are downloaded concurrently and cached better that way

File: src/styles/global.css [hunk 8]
```diff
@@ -0,0 +1,22 @@
+@import 'colors.css';
```
Comment (ryleu) [inline src/styles/global.css]: `@import` prevents stylesheets from being loaded concurrently, so just add both stylesheets in each html file

File: svelte.config.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+    kit: {
+		// adapter-auto only supports some environments, see https://svelte.dev/docs/kit/adapter-auto for a list.
+		// If your environment is not supported, or you settled on a specific environment, switch out the adapter.
+		// See https://svelte.dev/docs/kit/adapters for more information about adapters.
+		adapter: adapter({
+			// default options are shown. On some platforms
+			// these options are set automatically — see below
+			pages: '../static',
+			assets: '../static',
```
Comment (ryleu) [inline svelte.config.js]: should just be 'static', we want it to be self contained

**Review by ryleu — CHANGES_REQUESTED** [review]

**Review by ryleu — APPROVED** [review]
LGTM -- please squash when merging

**Pull request merged**